cdap-client/src/main/java/co/cask/cdap/client/util/RESTClient.java

```diff
@@ -98,11 +98,15 @@ private HttpResponse execute(HttpRequest request, int... allowedErrorCodes)
 
     int currentTry = 0;
     HttpResponse response;
+    int responseCode;
+    boolean allowUnavailable = ArrayUtils.contains(allowedErrorCodes, HttpURLConnection.HTTP_UNAVAILABLE);
+
     do {
       onRequest(request, currentTry);
       response = HttpRequests.execute(request, clientConfig.getDefaultRequestConfig());
+      responseCode = response.getResponseCode();
 
-      if (response.getResponseCode() != HttpURLConnection.HTTP_UNAVAILABLE) {
+      if (responseCode != HttpURLConnection.HTTP_UNAVAILABLE || allowUnavailable) {
```

Comment:
Yes, the `ServiceClientTestRun`. Because of this bug, it takes 1 mins for each test.